source/storing-data/archive-versions.rst

```diff
@@ -0,0 +1,226 @@
+   For a detailed description, see :doc:`provenance`.
+
+.. figure:: ../img/archive_versions/v1_prov_files.svg
+   :alt: Box and Arrow diagram of the provenance files in a v1 archive, as described above.
+
+V1 Archives look like this:
+
+.. figure:: ../img/archive_versions/v1-4_archive_fmt.svg
+   :alt: Box and Arrow diagram of a v1 archive, as described above.
+
+.. note::
+
+   V0 Archives do not capture provenance data. As a result, if a V0 artifact is 
+   an ancestor to a V1 (or greater) artifact, it is possible for the `action.yaml`
+   to list Artifact UUIDs which are not present in the `artifacts` directory.
+
+Version 2
+---------
+Released across QIIME 2 versions `2017.9 <https://github.com/qiime2/qiime2/releases/tag/2017.9.0>`_
+(`changelog <https://forum.qiime2.org/t/qiime-2-2017-9-release-is-now-live/1160>`__)
+and `2017.10 <https://github.com/qiime2/qiime2/releases/tag/2017.10.0>`_
+(`changelog <https://forum.qiime2.org/t/qiime-2-2017-10-release-is-now-live/1626>`__),
+the directory structure of this format is identical to v1,
+but the ``action.yaml`` file has changed.
+
+In `PR #333 <https://github.com/qiime2/qiime2/pull/333>`_, 
+the Version 2 ArchiveFormat adds an ``output-name`` key to the ``action`` section of ``action.yaml``
+(unless the action type is ``import``),
+assigning it the output name registered to the relevant action.
+Prior to this change, if one action returned multiple artifacts of the same :term:`Semantic Type`,
+it was not possible to differentiate between them using provenance alone.
+
+In `PR #348 <https://github.com/qiime2/qiime2/pull/348>`_,
+it adds provenance support for :term:`Pipelines <Pipeline>`,
+adding the ``alias-of`` key to the ``action`` section of ``action.yaml``.
+See description in :ref:`action-block` for details.
+
+Version 3
+---------
+
+Released in QIIME 2 version `2017.12 <https://github.com/qiime2/qiime2/releases/tag/2017.12.0>`_ (`changelog <https://forum.qiime2.org/t/qiime-2-2017-12-release-is-now-live/2308>`__),
+`PR #356 <https://github.com/qiime2/qiime2/pull/356>`_,
+the directory structure of this format is identical to v1 and v2.
+
+With this release, QIIME 2 Actions are able to take variadic arguments,
+allowing users to pass collections of Artifacts (``List`` s and ``Set`` s).
+A YAML representer has been added so that ``action.yaml`` can represent ``Set`` s of Artifact inputs.
+These will show up in ``action.yaml`` as custom ``!set`` tags.
+
+Version 4
+---------
+
+Released in QIIME 2 version `2018.4 <https://github.com/qiime2/qiime2/releases/tag/2018.4.0>`_ (`changelog <https://forum.qiime2.org/t/qiime-2-2018-4-release-is-now-live/3946>`__),
+`PR #387 <https://github.com/qiime2/qiime2/pull/387>`_,
+this format adds citations to the directory format,
+adds a ``transformers`` section to ``action.yaml``,
+and aligns the structure of ``environment:framework`` (also in ``action.yaml``)
+to match the structure of ``environment::plugins::<some_plugin>``.
```

Comment:
An overall observation - the line wrapping in this file is all over the place. I know some of these lines just can't be wrapped, but the ones that are are unpredictable - maybe if you get a chance when working on final edits you can unify? If not, no worries.

Reply:
I cleaned this up significantly. I've been experimenting with [semantic linefeeds](https://rhodesmill.org/brandon/2012/one-sentence-per-line/) in an effort to simplify future diffs. Splitting lines on sentences/clauses/ideas leaves a less consistent line length, but I like the overall effect.